### DISCUSSION TITLE
Mobile boilerplate React Native does not work

### DISCUSSION THREAD

**Discussion by miklyx** [DevSkillsHQ/discusssions#68] [Help] [answered]
Hi! 
Dependencies installation looks not so well. 

also, tries to open app in Android or IOs Emulator were not successful
`› Opening on Android...
CommandError: Required property 'android.package' is not found in the project app.json. This is required to open the app.
` - also "“emulator” cannot be opened because the developer cannot be verified."
ios also
`pening on iOS...
CommandError: Required property 'ios.bundleIdentifier' is not found in the project app.json. This is required to open the app.
`
when I try to run it like 'npm run ios' I see
`npm run ios    

> mobile-boilerplate-react-native@1.0.0 ios
> expo run:ios

📝  iOS Bundle Identifier Learn more: https://expo.fyi/bundle-identifier

✔ What would you like your iOS bundle identifier to be? … com.miklyx.mobileboilerplatereactnative1

✔ Created native project | gitignore skipped
✔ Updated package.json and added index.js entry point for iOS and Android
› Installing using npm
Using node to generate images. This is much slower than using native packages.
› Optionally you can stop the process and try again after successfully running `npm install -g sharp-cli`.

✔ Config synced
⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: Invalid `Podfile` file: 
[!] Invalid `RNGestureHandler.podspec` file: undefined method `exists?' for class File.

 #  from /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/node_modules/react-native-gesture-handler/RNGestureHandler.podspec:5
 #  -------------------------------------------
 #  
 >  isUserApp = File.exists?(File.join(__dir__, "..", "..", "node_modules", "react-native", "package.json"))
 #  if isUserApp
 #  -------------------------------------------
.

 #  from /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/ios/Podfile:14
 #  -------------------------------------------
 #    use_expo_modules!
 >    config = use_native_modules!
 #  
 #  -------------------------------------------

› Planning build

› 0 error(s), and 0 warning(s)

CommandError: Failed to build iOS project. "xcodebuild" exited with error code 70.
To view more error logs, try building the app with Xcode directly, by opening /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/ios/mobileboilerplatereactnative1.xcodeproj.

Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/ios/mobileboilerplatereactnative1.xcodeproj -configuration Debug -scheme mobileboilerplatereactnative1 -destination id=1C111E8B-56EB-45B2-AABC-E8CFF8C0D367

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

2024-02-15 17:45:20.571 xcodebuild[72286:13474819] Writing error result bundle to /var/folders/j0/ct4zzcy93_1fj5s2vzddy40r0000gp/T/ResultBundle_2024-15-02_17-45-0020.xcresult
xcodebuild: error: Unable to find a destination matching the provided destination specifier:
                { id:1C111E8B-56EB-45B2-AABC-E8CFF8C0D367 }

        Ineligible destinations for the "mobileboilerplatereactnative1" scheme:
                { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device, error:iOS 17.2 is not installed. To use with Xcode, first download and install the platform }
Build logs written to /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/.expo/xcodebuild.log`

tried with opened emulator:
`npm run ios    

> mobile-boilerplate-react-native@1.0.0 ios
> expo run:ios

📝  iOS Bundle Identifier Learn more: https://expo.fyi/bundle-identifier

✔ What would you like your iOS bundle identifier to be? … com.miklyx.mobileboilerplatereactnative1

✔ Created native project | gitignore skipped
✔ Updated package.json and added index.js entry point for iOS and Android
› Installing using npm
Using node to generate images. This is much slower than using native packages.
› Optionally you can stop the process and try again after successfully running `npm install -g sharp-cli`.

✔ Config synced
⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: Invalid `Podfile` file: 
[!] Invalid `RNGestureHandler.podspec` file: undefined method `exists?' for class File.

 #  from /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/node_modules/react-native-gesture-handler/RNGestureHandler.podspec:5
 #  -------------------------------------------
 #  
 >  isUserApp = File.exists?(File.join(__dir__, "..", "..", "node_modules", "react-native", "package.json"))
 #  if isUserApp
 #  -------------------------------------------
.

 #  from /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/ios/Podfile:14
 #  -------------------------------------------
 #    use_expo_modules!
 >    config = use_native_modules!
 #  
 #  -------------------------------------------

› Planning build

› 0 error(s), and 0 warning(s)

CommandError: Failed to build iOS project. "xcodebuild" exited with error code 70.
To view more error logs, try building the app with Xcode directly, by opening /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/ios/mobileboilerplatereactnative1.xcodeproj.

Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/ios/mobileboilerplatereactnative1.xcodeproj -configuration Debug -scheme mobileboilerplatereactnative1 -destination id=1C111E8B-56EB-45B2-AABC-E8CFF8C0D367

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

2024-02-15 17:45:20.571 xcodebuild[72286:13474819] Writing error result bundle to /var/folders/j0/ct4zzcy93_1fj5s2vzddy40r0000gp/T/ResultBundle_2024-15-02_17-45-0020.xcresult
xcodebuild: error: Unable to find a destination matching the provided destination specifier:
                { id:1C111E8B-56EB-45B2-AABC-E8CFF8C0D367 }

        Ineligible destinations for the "mobileboilerplatereactnative1" scheme:
                { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device, error:iOS 17.2 is not installed. To use with Xcode, first download and install the platform }
Build logs written to /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/.expo/xcodebuild.log
(base) mrashkovskiy@Kristinas-Air app % npm run ios

> mobile-boilerplate-react-native@1.0.0 ios
> expo run:ios

⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: Invalid `Podfile` file: 
[!] Invalid `RNGestureHandler.podspec` file: undefined method `exists?' for class File.

 #  from /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/node_modules/react-native-gesture-handler/RNGestureHandler.podspec:5
 #  -------------------------------------------
 #  
 >  isUserApp = File.exists?(File.join(__dir__, "..", "..", "node_modules", "react-native", "package.json"))
 #  if isUserApp
 #  -------------------------------------------
.

 #  from /Users/mrashkovskiy/Documents/js/data4life/data4-life-meal-roulette-app-mobile-h50jf-bh9ic0/app/ios/Podfile:14
 #  -------------------------------------------
 #    use_expo_modules!
 >    config = use_native_modules!
 #  
 #  -------------------------------------------`

Here's 'npm i' from terminal :
`npm install --force
npm WARN using --force Recommended protections disabled.
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.

added 2844 packages, and audited 2845 packages in 17s

153 packages are looking for funding
  run `npm fund` for details

110 vulnerabilities (1 low, 65 moderate, 37 high, 7 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.`

I've tried to reinstall node, cleared npm cache etc - no results.

Could I build this application from the scratch maybe , not using boilerplate?

**Comment by fenske** [comment, marked as answer]
Hi @miklyx. I'm Anton from Alva. Sorry for the slow reply here. It's alright to build your own application. We'll look into what's wrong with the boilerplate. Sorry for the inconvenience and let me know if you have any other questions.

**Reply by miklyx** [reply]
Thanks!
